client/src/components/Rating/RatingStars.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faStar as EmptyStar} from '@fortawesome/free-regular-svg-icons';
+import {faStar as SolidStar}  from '@fortawesome/free-solid-svg-icons';
+import './RatingStyles.css';
+
+export const RatingStars = () => {
+  let stars = new Array(5).fill(EmptyStar);
+  const [starSet, setStarSet] = useState(stars);
+  const [rating, setRating] = useState();
+
+  const changeStars = (rating) => {
+    stars = stars.map((star, index) => {
+      if(index <= rating) {
+        return SolidStar; 
+      } else {
+        return EmptyStar;
+      }
+    })
+    setStarSet(stars)
+  }
+
+  const saveOnServer = () => {
+    // console.log('Rating '+ ratingForSaving + " stars")
+    // console.log('not Saved') //comment: добавить запрос на сервер
+  }
+
+  const onStarClick = (e) => {
+    e.preventDefault();
+    const starId = e.currentTarget.id;
+    changeStars(+starId);
+    setRating(+starId+1)
+    saveOnServer()
+    console.log(rating)
+  }
+
+  const onHover = (e) => {
+    if(!rating) {
+      changeStars(e.target.id)
+    }
+  }
+
+  const onMouseLeave = () => {
+    if(!rating) {
+      changeStars(-1);
+    }
+  }
+
+  return (
+    <div>
+      <div className="Rating">
+        <ul className="Rating__list">
+          <li className="Rating__item">
+            <a className="Rating__link"><FontAwesomeIcon icon={starSet[0]} id="0" onMouseLeave={onMouseLeave} onMouseEnter={onHover} onClick={onStarClick} style={{color:'FDBF5A'}}/></a>
+          </li>
+          <li className="Rating__item">
+            <a  className="Rating__link"><FontAwesomeIcon icon={starSet[1]} id="1" onMouseLeave={onMouseLeave} onMouseEnter={onHover} onClick={onStarClick} style={{color:'FDBF5A'}}/></a>
+          </li>
```

Comment:
You can use array and loop through it. Like this:
```js
const stars = [1,2,3,4,5];

stars.map(star => 
    <li className="Rating__item" key={star}>
        <a  className="Rating__link"><FontAwesomeIcon icon={starSet[star]} id={star} onMouseLeave={onMouseLeave} onMouseEnter={onHover} onClick={onStarClick}/></a>
     </li>)
```

Reply:
I knew that you would say that)) I should have done this by myself. I did this already before.